### PR TITLE
Add e2e test for block settings menu

### PR DESCRIPTION
**Pull request by oandregal**
This PR adds a new e2e test for the plugins API: the `PluginBlockSettingsMenuItem` SlotFill.

The added plugin registers a new item in the block settings sidebar whose text is "My new plugin" and that, upon being clicked, outputs a "Block clicked" in the console.

### CODE REVIEW

**Comment by youknowriad** [comment]
did you omit commiting the test file? I only see changes to the plugin?

**Comment by nosolosw** [comment]
This is embarrasing :sweat_smile: I thought that I have done it, but it turns out that I didn't. Pushed it on 19431c4

**Review by youknowriad — APPROVED** [review]
LGTM 👍 Thanks for the update

**Comment by youknowriad** [comment]
Looks like it's breaking in travis though.

**Comment by aristath** [comment]
Rebased the PR, we no longer use Travis so let's see if tests will now pass

File: packages/e2e-tests/specs/editor/plugins/plugins-api.test.js [hunk 1]
```diff
@@ -156,4 +159,16 @@ describe( 'Using Plugins API', () => {
...
+
+	describe( 'Block Settings Menu Item', () => {
+		it( 'Should render a new item', async () => {
+			await insertBlock( 'List' );
+			await page.keyboard.type( 'one' );
+			await page.keyboard.press( 'Enter' );
+			await clickBlockToolbarButton( 'Options' );
+			await clickMenuItem( 'My new plugin' );
+
+			expect( console ).toHaveLoggedWith( 'Block clicked' );
```
Comment (oandregal) [inline packages/e2e-tests/specs/editor/plugins/plugins-api.test.js:171]: According to the report https://github.com/WordPress/gutenberg/pull/20357/checks?check_run_id=3251810969 it looks like the console.log [call](https://github.com/WordPress/gutenberg/pull/20357/files#diff-9087981e4092d9795696e561178cca7679249615973794c0c575ae2e471ac2b0R11) was not executed. I can see this working fine in interactive mode, though.

Comment (oandregal) [reply packages/e2e-tests/specs/editor/plugins/plugins-api.test.js:171]: Updated it to `toHaveLogged` locally and it also fails, so I wonder if I have to do anything special to test console messages.

**Pull request closed without merging**